### PR TITLE
Ask whether to proceed if map missing

### LINKED ISSUE #1246: Dont try download the map if player already has it
So I was doing some online testing of my coop missions. Let's forgtet for a bit about the other issue with coop replays.

But then when I have a local copy of my coop replay, which has all the information needed. Example:
```
{"uid": 9654895, "recorder": "speed2", "featured_mod": "coop", "launched_at": 1557605524.579191, "complete": true, "state": "PLAYING", "num_players": 2, "max_players": 4, "title": "speednoob only", "host": "FunkOff", "mapname": "scca_coop_a06", "map_file_path": "maps/scca_coop_a06.zip", "teams": {"1": ["FunkOff", "speed2"]}, "featured_mod_versions": {"1": 55, "2": 55, "3": 49, "4": 49, "5": 49, "6": 49, "7": 51, "8": 49, "9": 50, "10": 50, "11": 50, "12": 50, "13": 50, "14": 50, "15": 49, "16": 49, "17": 50, "18": 51, "19": 50, "20": 50, "21": 51, "22": 49}, "sim_mods": {}, "password_protected": false, "visibility": "PUBLIC", "command": "game_info", "game_end": 1557610797.2083693}
```
the client can't actually play it
```
java.util.concurrent.CompletionException: java.io.FileNotFoundException: http://content.faforever.com/faf/vault/maps/scca_coop_a06.zip
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
	at com.faforever.client.task.CompletableTask.lambda$new$1(CompletableTask.java:17)
	at javafx.base/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at javafx.base/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at javafx.base/javafx.event.Event.fireEvent(Unknown Source)
	at javafx.graphics/javafx.concurrent.EventHelper.fireEvent(Unknown Source)
	at javafx.graphics/javafx.concurrent.Task.fireEvent(Unknown Source)
	at javafx.graphics/javafx.concurrent.Task.setState(Unknown Source)
	at javafx.graphics/javafx.concurrent.Task$TaskCallable.lambda$call$2(Unknown Source)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$9(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: http://content.faforever.com/faf/vault/maps/scca_coop_a06.zip
	at java.base/jdk.internal.reflect.GeneratedConstructorAccessor101.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection$10.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getChainedException(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at com.faforever.client.map.DownloadMapTask.call(DownloadMapTask.java:55)
	at com.faforever.client.map.DownloadMapTask.call(DownloadMapTask.java:22)
	at javafx.graphics/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	... 1 more
Caused by: java.io.FileNotFoundException: http://content.faforever.com/faf/vault/maps/scca_coop_a06.zip
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(Unknown Source)
	at java.base/sun.net.www.protocol.http.HttpURLConnection.getHeaderField(Unknown Source)
	at java.base/java.net.URLConnection.getHeaderFieldLong(Unknown Source)
	at java.base/java.net.URLConnection.getContentLengthLong(Unknown Source)
	at java.base/java.net.URLConnection.getContentLength(Unknown Source)
	at com.faforever.client.map.DownloadMapTask.call(DownloadMapTask.java:51)
	... 6 more
```

It complains that the map isn't in the vault, correct?
Well its not there (since its my testing version), but the map is in my folder, so it should look there first before even trying to download it.

### PR DESCRIPTION
**Pull request by 1-alex98**
Fixes #1246

### CODE REVIEW

**Comment by 1-alex98** [comment]
@speed2CZ once this is merged I will force u to use the java client again XD

**Comment by 1-alex98** [comment]
if nobody dares to review i will just merge it just saying

File: src/main/java/com/faforever/client/game/GameService.java [hunk 1]
```diff
@@ -354,6 +358,25 @@ public void runWithReplay(Path path, @Nullable Integer replayId, String featured
...
+    AtomicReference<Boolean> proceed = new AtomicReference<>(false);
+    List<Action> actions = Arrays.asList(new Action(i18n.get("replay.ignoreMapNotFound"), event -> {
+          proceed.set(true);
+          userAnswered.countDown();
+        }),
+        new Action(i18n.get("replay.abortAfterMapNotFound"), event -> userAnswered.countDown()));
+    notificationService.addNotification(new ImmediateNotification(i18n.get("replay.mapDownloadFailed"), i18n.get("replay.mapDownloadFailed.wannaContinue"), Severity.WARN, actions));
+    userAnswered.await();
+    if (!proceed.get()) {
+      throw throwable;
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/game/GameService.java:397]: I'd throw the exception no matter whether the user proceeds or not.

**Review by Geosearchef — DISMISSED** [review]

**Comment by andreasley** [comment]
Actually, I think this patch might not be needed anymore, because https://github.com/FAForever/downlords-faf-client/pull/1561 should fix the underlying issue.

**Edit:** Well, there might still be circumstances where the map really isn't there anymore, so no reason not to merge.

**Review by andreasley — DISMISSED** [review]
Looks good to me.

File: src/main/java/com/faforever/client/game/GameService.java (hunk 1 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/game/GameService.java:397]: No u don't get the hack I am doing here right... Maybe I should log it that is a good idea but what u suggest is just not working

**Comment by 1-alex98** [comment]
@andreasley Yeah let's keep booth right?

**Comment by 1-alex98** [comment]
@andreasley btw u are allowed to give aprovals (reviews)

**Pull request merged**